### ISSUE TITLE
Need better python acme client docs/examples

### ISSUE THREAD

**Issue by optix2000** [certbot/certbot#2947]
The current documentation for the [Python acme client](http://letsencrypt.readthedocs.io/projects/acme/en/latest/index.html) is fairly lacking. The example client doesn't demonstrate how to set up challenges, and the docs for [`acme.challenges`](http://letsencrypt.readthedocs.io/projects/acme/en/latest/api/challenges.html) also does not give an example on how to set up challenges, nor does it provide enough description on how each class should be used.

This makes it difficult for someone to use the Python acme client to mint their own certs without either having to dig through the code for letsencrypt (certbot) or shelling out to certbot itself. This in turn makes the acme protocol harder to adopt.

Compare this with the example client in for the [Ruby acme client](https://github.com/unixcharles/acme-client/blob/master/README.md), which gives a full working client and an (arguably) easier to use API.
### TLDR

There should be a full end-to-end example for the Python acme client to make it easier to learn.

**Comment by kuba** [comment]
https://github.com/certbot/certbot/blob/master/acme/examples/example_client.py

**Comment by optix2000** [comment]
> The example client doesn't demonstrate how to set up challenges, and the docs for acme.challenges also does not give an example on how to set up challenges

As far as I can tell, the example client you linked is the same on in the docs. It's not a complete example as it doesn't show how to use challenges, nor is it capable of minting certs as it mentions on line 47.

I'm asking for a complete example end-to-end that shows how to use challenges.

**Comment by kuba** [comment]
I suggest you have a look at https://github.com/kuba/simp_le for reference implementation.

**Comment by bmw** [comment]
I agree that we can significantly improve things here so I'm adding this to our wishlist.

**Comment by bmw** [comment]
One thing this example should do is show how to reuse an existing account.

**Comment by bmw** [comment]
I think this has gotten a lot better with PRs like https://github.com/certbot/certbot/pull/5494, but many of the suggestions in this thread have not been done yet.

**Comment by Ornias1993** [comment]
I can definately confirm the examples are still lacking. Primarily in the DNS-01 side of things, which is slowly becoming somewhat of the standard/default due to support for wildcards (which HTTP-01 does not).

**Comment by Ornias1993** [comment]
Needs updates by the devs, not us keeping it open endlessly.
